bitnami/kafka/templates/jmx-metrics-svc.yaml

```diff
@@ -18,28 +19,16 @@ metadata:
     {{- end }}
   {{- end }}
 spec:
-  type: {{ .Values.metrics.jmx.service.type }}
-  {{- if eq .Values.metrics.jmx.service.type "LoadBalancer" }}
-  {{- if .Values.metrics.jmx.service.loadBalancerIP }}
-  loadBalancerIP: {{ .Values.metrics.jmx.service.loadBalancerIP }}
-  {{- end }}
-  {{- if .Values.metrics.jmx.service.loadBalancerSourceRanges }}
-  loadBalancerSourceRanges: {{- toYaml .Values.metrics.jmx.service.loadBalancerSourceRanges | nindent 4 }}
-  {{- end }}
-  {{- end }}
-  {{- if and (eq .Values.metrics.jmx.service.type "ClusterIP") .Values.metrics.jmx.service.clusterIP }}
+  type: ClusterIP
+  sessionAffinity: {{ .Values.metrics.jmx.service.sessionAffinity }}
+  {{- if .Values.metrics.jmx.service.clusterIP }}
   clusterIP: {{ .Values.metrics.jmx.service.clusterIP }}
   {{- end }}
   ports:
     - name: http-metrics
-      port: {{ .Values.metrics.jmx.service.port }}
+      port: {{ .Values.metrics.jmx.service.ports.metrics }}
       protocol: TCP
       targetPort: metrics
-      {{- if and (or (eq .Values.metrics.jmx.service.type "NodePort") (eq .Values.metrics.jmx.service.type "LoadBalancer")) (not (empty .Values.metrics.jmx.service.nodePort)) }}
-      nodePort: {{ .Values.metrics.jmx.service.nodePort }}
-      {{- else if eq .Values.metrics.jmx.service.type "ClusterIP" }}
-      nodePort: null
-      {{- end }}
```

Comment:
I'm sorry but I don't understand these changes. Why forcing JMX metrics to be a ClusterIP service?

Reply:
Because it's a very bad practice (high security risk) to expose metrics that describe the internal behaviour of the apps outside the cluster.

Reply:
Understood, thanks for the clarification